src/Algebra/Graph.hs

```diff
@@ -464,6 +479,13 @@ hasEdge u v = (edge u v `isSubgraphOf`) . induce (`elem` [u, v])
 -- @
 vertexCount :: Ord a => Graph a -> Int
 vertexCount = Set.size . vertexSet
+{-# INLINE[1] vertexCount #-}
+
+{-# RULES "vertexCount/Int" vertexCount = vertexIntCount #-}
+
+-- | Specialized version of 'vertexcount' for graphs with vertices of type 'Int'.
```

Comment:
We probably don't need to export it, since the user will always get it automatically?

Reply:
Yep you are right, no need to export :) That is the tactic used here: https://github.com/haskell/fgl/blob/master/Data/Graph/Inductive/PatriciaTree.hs#L164

Reply:
Aha, good to know that competing libraries are also specialising to `Int` :)